proposals/lip-0053.md

```diff
@@ -252,178 +479,192 @@ crossChainUpdateTransactionParams = {
-  * <code>[appendToInboxTree][lip-0045#appendToInboxTree](ccu.sendingChainID, serializedCCM)</code>, where `serializedCCM` is the serialization of `ccm` using [`crossChainMessageSchema`][lip-0049#ccmSchema].
-  * <code>[apply][lip-0049#apply](ccu, ccm)</code>.
-* Continue the processing by executing all steps in the ["Common Processing" section](#common-processing).
+```python
+def verify(trs: Transaction) -> None:
+    # Only the mainchain can send this command.
+    if trs.params.sendingChainID != MAINCHAIN_ID:
+         raise Exception("Only the mainchain can send a sidechain cross-chain update.")
+    if len(trs.params.certificate) > 0:
+        verifyCertificate(trs)
+    if len(trs.params.activeValidatorsUpdate) > 0 or trs.params.newCertificateThreshold > 0:
+        verifyValidatorsUpdate(trs)
+    if trs.params.inboxUpdate is not empty: # An empty object has all properties set to their default values.
+        verifyChannel(trs)
+```
 
-#### Common Processing
+##### Execution
 
-For CCU transactions posted on the mainchain or on sidechains, once the specific execution steps described above have been completed, the following is done in the given order:
+```python
+def execute(trs: Transaction) -> None:
+    # Verify certificate signature. We do it here because if it fails, the transaction fails rather than being invalid.
+    verifyCertificateSignature(trs)
+    # Process cross-chain messages in inbox update.
+    crossChainMessages = trs.params.inboxUpdate.crossChainMessages
+
+    # First process basic checks for all CCMs.
+    for ccmBytes in crossChainMessages:
+        # Verify general format. Past this point, we can access ccm root properties.
+        try:
+            validateFormat(ccm)
+            # The CCM must come be directed to the sidechain, unless it was bounced on the mainchain.
+            if ownChainAccount.ID != ccm.receivingChainID and ccm.status != CCM_STATUS_CHANNEL_UNAVAILABLE:
+                raise Exception("CCM is not directed to the sidechain.")
+            elif ownChainAccount.ID != ccm.sendingChainID and ccm.status == CCM_STATUS_CHANNEL_UNAVAILABLE:
+                raise Exception("Bounced CCM was not coming from the sidechain.")
+        except:
+            terminateChain(trs.params.sendingChainID)
+            # In this case, we do not even update the chain account with the new certificate.
+            return
+    
+    for ccmBytes in crossChainMessages:
+        try:
+            ccm = decode(crossChainMessageSchema, ccmBytes)
+            # Modules can verify the CCM. In particular, this calls the interoperability verifyCrossChainMessage defined above.
+            for each module mdl for which verifyCrossChainMessage exists:
+                mdl.verifyCrossChainMessage(trs, ccm)
+        except:
+            # Chain was already terminated.
+            if terminatedStateAccount(chainID) exists:
+                continue
```

Comment:
Why do we continue here as opposed to the execute logic for the other command? Would we still want to emit a message what happened to the CCM?